presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -166,7 +167,13 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                         ImmutableList.of(new CountConstantOptimizer()),
                         ImmutableSet.of(new SimplifyCountOverConstant())),
                 new WindowFilterPushDown(metadata), // This must run after PredicatePushDown and LimitPushDown so that it squashes any successive filter nodes and limits
-                new MergeWindows(),
+                new IterativeOptimizer(
+                                stats,
+                                ImmutableList.of(new MergeWindows()),
```

Comment:
Old `MergeWindows` depends on `UnaliasSymbolReferences` and `PruneIdentityProjections` rules. I've done this similarly in https://github.com/prestodb/presto/pull/7300/files#diff-4ae44d30e3c35f6a5fa5e5c03653f974R176.